samples/NativeCatalog/android/app/src/main/java/com/pspdfkit/nativecatalog/MainApplication.java

```diff
@@ -0,0 +1,76 @@
+package com.pspdfkit.nativecatalog;
+
+import android.app.Application;
+import android.content.Context;
+
+import com.facebook.react.PackageList;
+import com.facebook.react.ReactApplication;
+import com.facebook.react.ReactNativeHost;
+import com.facebook.react.ReactPackage;
+import com.facebook.soloader.SoLoader;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.List;
+
+public class MainApplication extends Application implements ReactApplication {
+
+    private final ReactNativeHost mReactNativeHost =
+        new ReactNativeHost(this) {
+            @Override
+            public boolean getUseDeveloperSupport() {
+                return BuildConfig.DEBUG;
+            }
+
+            @Override
+            protected List<ReactPackage> getPackages() {
+                @SuppressWarnings("UnnecessaryLocalVariable")
+                List<ReactPackage> packages = new PackageList(this).getPackages();
+                // Packages that cannot be autolinked yet can be added manually here, for example:
+                packages.add(new CustomPdfViewPackage());
```

Comment:
Is this class auto-generated? This comment and the code it is referring to sounds pretty generic (and not really helpful IMO). I'm not too fond of RN. Would there a way for RN to "autolink" custom packages at all? Or is this always necessary when creating a binding like the example does?

Reply:
Yes it is autogenerated. Packages that are supplied by external dependencies are autolinked, but ones you create in your own project always need to be added manually.